### ISSUE TITLE
CI smoke test produces no app coverage data (0 .gcda) — combined coverage is misleading

### ISSUE THREAD

**Issue by davidguyomarch** [davidguyomarch/Sankore#230]
## Problem

The Linux CI (`build-linux.yml`) is meant to report three coverage numbers:
1. **App coverage** — the release binary run under a smoke scenario (`Open-Sankore --quit-after=10 --smoke-scenario`)
2. **Unit test coverage** — the unit test binary
3. **Combined** — the union of the two

On the latest successful master run (#33544543598, commit `1acbc70`), the app coverage step reports:

```
App smoke test completed
Found 0 .gcda files
...
Found 0 app .gcda files
No .gcda files found — app coverage not available
```

So **no app coverage is captured at all**. The "Combined" number then falls back to the unit-test tracefile:

```
COMBINED COVERAGE (App + Unit Tests)
lines......: 87.7% (3794 of 4325 lines)
```

This is identical to the unit-test-only number because the app side is empty.

## Why this is misleading

The advertised **87.7% "combined"** is not what it appears to be. Its denominator (4325 lines) is only the ~25 self-contained files compiled into the unit test binary — NOT the whole application. The metric that would answer "how much of the release binary's code is exercised" (release lines executed / total project lines, via `lcov --capture --no-external`) is currently **not measured**, because the smoke run writes zero `.gcda` files.

Net effect: the project looks like it has ~88% coverage "globally", when in reality the whole-application coverage is unknown.

## Notes on the smoke test itself

The smoke run does NOT crash — the log shows it completes the scenario (adds pages, draws a stroke, exports a PDF) and prints `App smoke test completed`. So the missing `.gcda` is not a crash-before-flush problem.

`src/core/main.cpp` already tries to flush gcov before exit in the `--quit-after` watchdog thread:

```cpp
extern "C" void __gcov_dump(void) __attribute__((weak));
...
std::thread([timeout]() {
    std::this_thread::sleep_for(std::chrono::seconds(timeout));
#ifndef _WIN32
    if (__gcov_dump)
        __gcov_dump();
#endif
    _exit(0);
}).detach();
```

Since the guarded `__gcov_dump()` is followed by `_exit(0)` (which bypasses the normal atexit gcov flush), the whole mechanism depends on the weak `__gcov_dump` symbol actually resolving at runtime. Likely root causes to investigate:

- The weak `__gcov_dump` symbol resolves to null at runtime (not linked from libgcov in this build), so the flush is silently skipped, and `_exit(0)` then writes nothing.
- Or the `.gcda` files are written relative to the compile-time object path (`build/linux/release/objects/`) while the smoke step's `find .` runs from a different working directory, so they are simply not found.

## Suggested fixes to explore

1. Verify `__gcov_dump` is non-null at runtime under the CI build flags; if it is null, ensure the app links libgcov so the symbol resolves (or call `__gcov_flush`/`__gcov_dump` from a strongly-linked TU).
2. Confirm where `.gcda` files land and align the `find`/`lcov --directory` path with it.
3. Prefer a clean shutdown (`qApp->quit()` + return from `exec()`) over `_exit(0)` where possible, so the standard gcov atexit flush runs.

## Acceptance criteria

- The smoke step reports a non-zero `.gcda` count.
- The CI job summary shows three distinct numbers (app / unit / combined), with the combined denominator reflecting the full application, not just the unit-tested files.

Refs #66
